src/gemmA.cc

```diff
@@ -83,7 +83,8 @@ void gemmA(
                 for (int64_t i = 0; i < B.mt(); ++i)
                     bcast_list_B.push_back(
                         {i, k, {A.sub( 0, A.mt()-1, i, i )}} );
-                B.template listBcast<target>( bcast_list_B, layout, k );
+                int tag_k = k;
```

Comment:
The var `k` is `int64_t` but `tag_k` is `int`.

Reply:
I checked, and the routine takes an `int`. So ignore the previous comment.

Reply:
Generally, we use `int` for MPI tags, MPI ranks, and GPU devices, because the underlying libraries use `int` for those things. We wouldn't gain anything by using `int64_t`.

Hmm... that does point out that `k` can't be too huge, as `MPI_TAG_UB` is only guaranteed to be at least 32767. For nb = 200, that allows n = 6,553,400. I would guess that most MPI impl have larger MPI_TAG_UB, though.

Reply:
Good to know. I did not know about `MPI_TAG_UB`